src/main/java/com/xebialabs/overcast/host/CachedLibvirtHost.java

```diff
@@ -86,7 +86,7 @@
         List<Filesystem> filesystemMappings, List<String> copySpec) {
         super(libvirt, baseDomainName, ipLookupStrategy, networkName, startTimeout, bootDelay, filesystemMappings);
         this.provisionUrl = checkNotNullOrEmpty(provisionUrl, "provisionUrl");
-        this.provisionCmd = checkNotNullOrEmpty(provisionCmd, "provisionCmd");
+        this.provisionCmd = checkNotNullOrEmpty(provisionCmd, "provisionCmd").trim();
```

Comment:
We trim once 'checkNotNullOrEmpty' check has passed, at this point we are sure that trim operation will not throw a null pointer exception.